force-app/main/default/triggers/AccountTrigger.trigger

```diff
@@ -0,0 +1,81 @@
+/* Account trigger should do the following:
+* 1. Set the account type to prospect.
+* 2. Copy the shipping address to the billing address.
+* 3. Set the account rating to hot.
+* 4. Create a contact for each account inserted.  
+Question 1
+    * Account Trigger
+    * When an account is inserted change the account type to 'Prospect' if there is no value in the type field.
+    * Trigger should only fire on insert.
+Question 2
+    * Account Trigger
+    * When an account is inserted copy the shipping address to the billing address.
+    * BONUS: Check if the shipping fields are empty before copying.
+    * Trigger should only fire on insert.
+Question 3
+    * Account Trigger
+	* When an account is inserted set the rating to 'Hot' if the Phone, Website, and Fax ALL have a value.
+    * Trigger should only fire on insert.
+Question 4
+    * Account Trigger
+    * When an account is inserted create a contact related to the account with the following default values:
+    * LastName = 'DefaultContact'
+    * Email = 'default@email.com'
+    * Trigger should only fire on insert. 123
+*/
+
+
+trigger AccountTrigger on Account (before insert, after insert) {
+
+    if (Trigger.isInsert) {
+        // Before Insert Logic
+        if (Trigger.isBefore) {
+            for (Account acct : Trigger.new) {
+                // Set Account Type to 'Prospect' if not provided
+                if (acct.Type == null) {
+                    acct.Type = 'Prospect';
+                }
+
+                // Copy Shipping Address to Billing Address if Shipping fields are populated
+                if (!String.isBlank(acct.ShippingStreet) ||
+                    !String.isBlank(acct.ShippingCity) ||
+                    !String.isBlank(acct.ShippingState) ||
+                    !String.isBlank(acct.ShippingPostalCode) ||
+                    !String.isBlank(acct.ShippingCountry)) {
```

Comment:
You probably want to check the target is not blank, not the source. Copying over from blank values doesn't really matter, but if the values are supplied, you probably do not want to overwrite them.